File: src/swingjs/plaf/JSTextAreaUI.java
package swingjs.plaf;

import java.awt.Event;

import jsjava.awt.Dimension;
import jsjava.awt.Insets;
import jsjavax.swing.text.JTextComponent;
import swingjs.api.DOMNode;

/**
 * SWingJS implementation of stateful user interface for buttons. 
 * Modeled after javax.swing.plaf.basic.BasicButtonUI.java (commented out below).
 * 
 * @author Bob Hanson
 *
 */
public class JSTextAreaUI extends JSTextUI {

	/**
	 * the radio or check-box or simple button
	 * 
	 */
	protected DOMNode domBtn;

	@Override
	protected DOMNode updateDOMNode() {
		if (domNode == null) {
			allowBackground = false;
			domBtn = focusNode = enableNode = textNode = valueNode = domNode = newDOMObject("textarea", id);
			DOMNode.setStyles(domNode, "resize", "none");
			setDataUI(domNode);
			bindJSEvents(domNode, "keydown keypress keyup focusout", Event.KEY_PRESS, false);
			addJQueryFocusCallbacks();
		}
		textListener.checkDocument();
		setCssFont(
				DOMNode.setAttr(domNode, "innerHTML", getComponentText()),
				c.getFont());
		if (!editable)
			DOMNode.setAttr(domNode, "readOnly", "true");
		return domNode;
	}

	private Insets myInsets = new Insets(0, 0, 5, 5); 
	@Override
	public Insets getInsets() {
		return myInsets;
	}
	
	@Override
	protected Dimension getCSSAdjustment(boolean addingCSS) {
		return (addingCSS ? new Dimension(-5, -12) : new Dimension(0, 0)); 
		// total hack -12 is to see full vertical scrollbar (Boltzmann)
	}

	@Override
	protected String getPropertyPrefix() {
		return "TextArea";
	}
	
}

File: site/swingjs/j2s/swingjs/plaf/JSTextAreaUI.js
Clazz.declarePackage ("swingjs.plaf");
Clazz.load (["swingjs.plaf.JSTextUI"], "swingjs.plaf.JSTextAreaUI", ["swingjs.api.DOMNode"], function () {
c$ = Clazz.decorateAsClass (function () {
this.domBtn = null;
Clazz.instantialize (this, arguments);
}, swingjs.plaf, "JSTextAreaUI", swingjs.plaf.JSTextUI);
Clazz.overrideMethod (c$, "getDOMObject", 
function () {
if (this.domNode == null) {
this.updateHandler.checkDocument ();
this.domBtn = this.focusNode = this.enableNode = this.textNode = this.domNode = this.createDOMObject ("textarea", this.id, []);
this.bindMouse (this.domNode);
if ((this.c).isEditable ()) {
this.bindKeys (this.domNode);
this.setFocusable ();
}}this.setCssFont (swingjs.api.DOMNode.setAttr (this.domNode, "innerHTML", this.getComponentText ()), this.c.getFont ());
if (!this.editable) swingjs.api.DOMNode.setAttr (this.domNode, "readOnly", "true");
return this.domNode;
});
});
